assembl/views/api2/discussion.py

```diff
@@ -667,6 +667,143 @@ def multi_module_csv_export(request):
+
+    start, end, interval = get_time_series_timing(request)
+    has_anon = asbool(request.GET.get('anon', False))
+
+    posts = db.query(m.Post
+        ).filter(m.Post.discussion_id == discussion.id
+        ).filter(m.Post.creation_date > start
+        ).filter(m.Post.creation_date < end
+        ).order_by(m.Post.id).all()
+
+    sentiment_counts = db.query(m.Post.id, m.SentimentOfPost.actor_id, m.SentimentOfPost.type).join(m.SentimentOfPost).filter(
+        m.Post.id.in_([post.id for post in posts]),
+        m.SentimentOfPost.tombstone_condition()
+    ).group_by(m.Post.id, m.SentimentOfPost.actor_id, m.SentimentOfPost.type).all()
+
+    # Sort sentiments for better access depending of post/user
+    sentiments_given_by_user = defaultdict(lambda: defaultdict(int))
+    sentiments_received_by_post = defaultdict(lambda: defaultdict(int))
+
+    for sentiment in sentiment_counts:
+        sentiments_received_by_post[sentiment[0]][sentiment[2]] += 1
+        sentiments_given_by_user[sentiment[1]][sentiment[2]] += 1
+
+    users = {}
+
+    users_in_discussion = db.query(m.User).join(m.AgentStatusInDiscussion, m.AgentStatusInDiscussion.profile_id == m.User.id).filter(
+        m.AgentStatusInDiscussion.discussion_id == discussion.id).order_by(m.User.id).all()
+
+    for user in users_in_discussion:
+        # Sort out ppl who couldn't have connected during the required period
+        if (user.first_visit and (user.creation_date > end or user.first_visit > end)) or (user.last_visit and user.last_visit < start):
+            continue
+
+        username = user.username_p.encode('utf-8') if user.username_p else ''
+
+        users[user.id] = {
+                NAME: user.name.encode('utf-8') if not has_anon else user.anonymous_name(),
+                EMAIL: user.get_preferred_email(has_anon).encode('utf-8'),
+                USERNAME: username if not has_anon else user.anonymous_username(),
+                CREATION_DATE: user.creation_date.strftime('%Y-%m-%d %H:%M:%S') if user.creation_date else '',
+                FIRST_VISIT: user.first_visit.strftime('%Y-%m-%d %H:%M:%S') if user.first_visit else '',
+                LAST_VISIT: user.last_visit.strftime('%Y-%m-%d %H:%M:%S') if user.last_visit else '',
+                AGREE_GIVEN: sentiments_given_by_user[user.id][LIKE_SENTIMENT],
+                DISAGREE_GIVEN: sentiments_given_by_user[user.id][DISLIKE_SENTIMENT],
+                DONT_UNDERSTAND_GIVEN: sentiments_given_by_user[user.id][DONT_UNDERSTAND_SENTIMENT],
+                MORE_INFO_GIVEN: sentiments_given_by_user[user.id][MORE_INFO_SENTIMENT],
+                AGREE_RECEIVED: 0,
+                DISAGREE_RECEIVED: 0,
+                DONT_UNDERSTAND_RECEIVED: 0,
+                MORE_INFO_RECEIVED: 0,
+                TOP_POSTS: 0,
+                TOP_POST_REPLIES: 0,
+                POSTS: 0,
+                POSTS_REPLIES: 0,
+                TOTAL_POSTS: 0,
+                THEMATICS: []
```

Comment:
Please change this to `Idea`, we're trying to move away from the name `Thematic`, as this object type no longer exists in backend (and will be deprecated on front-end). Also, I would no only add the Idea title as you've done, I would also add the `idea_id`, so that consultants can create pivot tables based on that column.